packages/ember-data/lib/adapters/rest-adapter.js

```diff
@@ -838,3 +838,15 @@ function endsWith(string, suffix) {
     return string.endsWith(suffix);
   }
 }
+
+if (Ember.platform.hasPropertyAccessors) {
+  Ember.defineProperty(RestAdapter, 'maxUrlLength', {
+    enumerable: false,
+    get: function() {
+      Ember.deprecate('maxUrlLength has been deprecated (wrong casing). You should use maxURLLength instead.');
+      return this.maxURLLength;
+    }
```

Comment:
Lets also add a setter otherwise a type error will get thrown if a user tires to extend an adapter using `maxUrlLength`.